presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -142,6 +142,19 @@ std::shared_ptr<connector::ColumnHandle> toColumnHandle(
         toRequiredSubfields(hiveColumn->requiredSubfields));
   }
 
+  if (auto icebergColumn =
```

Comment:
This function is becoming too long. Also the content for Iceberg is almost the same as Hive. Would it be possible to extract some sub functions? 
Now, let's see if this is possbile. The following values need to be gotten from TableLayoutHandle 
1. partitionColumns
    - hiveLayout->partitionColumns is List<HiveColumnHandle> partitionColumns
    -  icebergLayout>partitionColumns is List<IcebergColumnHandle> partitionColumns;
    - HiveColumnHandle and IcebergColumnHandle both implements BaseHiveColumnHandle. Can we move the partitionColumns field to BaseHiveTableLayoutHandle?
2. domains and remainingPredicate
    - domainPredicate and remainingPredicate are already in BaseHiveTableLayoutHandle, so we can get it directly from the base class

3. dataColumns
    - it's both `List<Column>` in IcebergTableLayoutHandle and HiveTableLayoutHandle. 
    - However the `Column` class is in presto-hive-metastore while BaseHiveTableLayoutHandle is in presto-hive-common, so we can't directly move the dataColumns field or the getter function to BaseHiveTableLayoutHandle.
    -  We can pass the List<Column> as a parameter to the extracted function
4. tableHandle's schemaName and tableName
    - HiveTableHandle and IcebergTableHandle both implements ConnectorTableHandle, and both have schemaName and tableName. I also checked other connector TableHandles and they mostly have schemaName and tableName. I think we can make ConnectorTableHandle contain the getters of these names. @tdcmeehan Do you see any problem in doing that?
    - If modifying ConnectorTableHandle is not favored, we can just pass schemaName and tableName to the extracted method
4. tableParameters 
    - Iceberg doesn't have this, just pass it in as a map parameter.
    
```
if (auto hiveLayout =
          std::dynamic_pointer_cast<const protocol::HiveTableLayoutHandle>(
              tableHandle.connectorTableLayout)) {
    List<Column> dataColumns = hiveLayout->dataColumns;
    return toHiveTableHandle(hiveLayout, dataColumns, tableHandle, tableParameters);
}
if (auto icebergLayout =
          std::dynamic_pointer_cast<const protocol::IcebergTableLayoutHandle>(
              tableHandle.connectorTableLayout)) {
    List<Column> dataColumns = icebergLayout->dataColumns;
    return toHiveTableHandle(icebergLayout, dataColumns, tableHandle)

```

Reply:
> HiveTableHandle and IcebergTableHandle both implements ConnectorTableHandle, and both have schemaName and tableName. I also checked other connector TableHandles and they mostly have schemaName and tableName. I think we can make ConnectorTableHandle contain the getters of these names. @tdcmeehan Do you see any problem in doing that?

Can we just put it in a common super type in `presto-hive-common`?

Reply:
> Can we just put it in a common super type in presto-hive-common?

@tdcmeehan Did you mean creating a HiveConnectorTableHandle interface in presto-hive-common and make HiveTableHandle, IcebergTableHandle and other Hive related table handles, e.g. Hudi, Delta all implement it?

Reply:
Yes. Actually, `presto-hive-common` is probably misnamed, it's more like `presto-lake-connector-common`.

Reply:
This was done but the name was not changed yet. We can rename it later.